### PR TITLE
Internals are kept on as long as any breathing tool is on

### LINKED ISSUE #27643: Internals toggling/check is bugged when wearing both a breath mask and hardsuit helmet
## Description

Putting on a hardsuit helmet while breathing from a gas mask turns internals off

Removing a hardsuit helmet while also wearing a gas mask turns internals off AND it won't turn back on again (reporting that no breath tool is being worn), until the mask is removed and re-equipped or the helmet is deployed again

The two problems are
-connecting a new breath tool always resets internals which might be reasonable when swapping one gas mask for another but not when adding a helmet on top of a gas mask ((or a gas mask under a helmet... somehow. but inventory layering is a whole other topic))
-when removing a breath tool the game presumably does not check if any other breath tools are left before updating the component with "no breath tool is being worn" (probably only mask slot items could be breath tools when that was coded so this scenario was not possible)

This is currently just a peculiar quirk since probably no one bothers wearing a mask with a hardsuit. It will be more annoying when there are species that need internals even without a hardsuit, such as vox; or if toxic atmospheres at room pressure/temp somehow become more common

**Reproduction**

-Spawn a Death Squad Agent (which conveniently comes with both a deployed helmet, an equipped gas mask. Conveniently, as of today, breath tools equipped on a mob at spawn start already active)
-Toggle helmet off
... Internals turn off, despite the gas mask still being worn
-Toggle internals on 
... Internals do not turn on
-Remove and re-equip gas mask, then turn internals on
-Toggle helmet on
... Internals turn off again

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Before the change:
Taking off your helmet or mask would toggle off internals even if you had both on.

After the change:
If you have both a mask and helmet on with internals turned on, and you un-equip either, internals will stay on until you take off both.

## Why / Balance

Fixes #27643 and #28567

## Technical details

InternalsComponent now uses a HashSet of breathing tools rather than just tracking a single breathing tool, all other code adjusted to account for the possibility of multiple breathing tools and the removal of the correct tool.

Uh... it... it kind of feels overkill to use a HashSet when the largest it can be at the moment is... two entities...? but yeah

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/58439124/de6c86f3-b8fa-46b6-9b89-2939c7282cad

## Breaking changes

InternalsComponent `public EntityUid? BreathToolEntity;` changed to `HashSet<EntityUid> BreathTools { get; set; } = new();`

**Changelog**

:cl:
- tweak: Internals are no longer toggled off if you take your helmet off but still have a gas mask on and vice versa.

### CODE REVIEW

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 1]
```diff
@@ -44,7 +45,7 @@ public override void Initialize()
 
     private void OnStartingGear(EntityUid uid, InternalsComponent component, ref StartingGearEquippedEvent args)
     {
-        if (component.BreathToolEntity == null)
+        if (!component.BreathTools.Any())
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:48]:
```suggestion
        if (component.BreathTools.Count == 0)
```

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 2]
```diff
@@ -111,7 +112,7 @@ private void OnStartingGear(EntityUid uid, InternalsComponent component, ref Sta
         }
 
         // If they're not on then check if we have a mask to use
-        if (internals.BreathToolEntity is null)
+        if (!internals.BreathTools.Any())
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:115]:
```suggestion
        if (internals.BreathTools.Count == 0)
```

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 3]
```diff
@@ -178,28 +179,22 @@ private void OnInhaleLocation(Entity<InternalsComponent> ent, ref InhaleLocation
...
-        ent.Comp.BreathToolEntity = null;
+        ent.Comp.BreathTools.Remove(toolEntity);
 
-        if (TryComp(old, out BreathToolComponent? breathTool))
-        {
-            _atmos.DisconnectInternals(breathTool);
+        if (TryComp(toolEntity, out BreathToolComponent? breathTool))
+            _atmos.DisconnectInternals(toolEntity, breathTool);
+
+        if (!ent.Comp.BreathTools.Any())
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:189]:
```suggestion
        if (ent.Comp.BreathTools.Count == 0)
```

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 4]
```diff
@@ -217,7 +212,7 @@ public void DisconnectTank(InternalsComponent? component)
 
     public bool TryConnectTank(Entity<InternalsComponent> ent, EntityUid tankEntity)
     {
-        if (ent.Comp.BreathToolEntity is null)
+        if (!ent.Comp.BreathTools.Any())
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:215]:
```suggestion
        if (ent.Comp.BreathTools.Count == 0)
```

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 5]
```diff
@@ -236,14 +231,14 @@ public bool AreInternalsWorking(EntityUid uid, InternalsComponent? component = n
...
-        return TryComp(component.BreathToolEntity, out BreathToolComponent? breathTool)
+        return TryComp(component.BreathTools.FirstOrNull(), out BreathToolComponent? breathTool)
             && breathTool.IsFunctional
             && HasComp<GasTankComponent>(component.GasTankEntity);
     }
 
     private short GetSeverity(InternalsComponent component)
     {
-        if (component.BreathToolEntity is null || !AreInternalsWorking(component))
+        if (!component.BreathTools.Any() || !AreInternalsWorking(component))
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:241]:
```suggestion
        if (component.BreathTools.Count == 0 || !AreInternalsWorking(component))
```

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 6]
```diff
@@ -12,6 +12,7 @@
 using Content.Shared.Verbs;
 using Robust.Shared.Containers;
 using Robust.Shared.Utility;
+using System.Linq;
```
Comment (Tayrtahn) [inline Content.Server/Body/Systems/InternalsSystem.cs:15]:
```suggestion
```

**Review by Tayrtahn — COMMENTED** [review]
I don't have time to do a full review right now, but I'm gonna nitpick about using `Any` instead of checking `Count`.

File: Content.Server/Body/Systems/InternalsSystem.cs [hunk 7]
```diff
@@ -178,28 +178,22 @@ private void OnInhaleLocation(Entity<InternalsComponent> ent, ref InhaleLocation
...
     public void ConnectBreathTool(Entity<InternalsComponent> ent, EntityUid toolEntity)
     {
-        if (TryComp(ent.Comp.BreathToolEntity, out BreathToolComponent? tool))
-        {
-            _atmos.DisconnectInternals(tool);
-        }
-
-        ent.Comp.BreathToolEntity = toolEntity;
+        ent.Comp.BreathTools.Add(toolEntity);
         _alerts.ShowAlert(ent, ent.Comp.InternalsAlert, GetSeverity(ent));
```
Comment (AJCM-git) [inline Content.Server/Body/Systems/InternalsSystem.cs:197]: shouldnt this be a tryadd?

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.BreathTool.cs [hunk 8]
```diff
@@ -12,17 +12,17 @@ private void InitializeBreathTool()
 
     private void OnBreathToolShutdown(EntityUid uid, BreathToolComponent component, ComponentShutdown args)
     {
-        DisconnectInternals(component);
+        DisconnectInternals(uid, component);
     }
 
-    public void DisconnectInternals(BreathToolComponent component)
+    public void DisconnectInternals(EntityUid uid, BreathToolComponent component)
```
Comment (AJCM-git) [inline Content.Server/Atmos/EntitySystems/AtmosphereSystem.BreathTool.cs:18]: Entity<>

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/Body/Systems/InternalsSystem.cs (hunk 7 above)
Comment (Plykiya) [reply Content.Server/Body/Systems/InternalsSystem.cs:197]: that doesn't seem to be a function for Hashsets...

Comment (Plykiya) [reply Content.Server/Body/Systems/InternalsSystem.cs:197]: and toolEntity doesn't really seem to have a chance to be null?

Comment (AJCM-git) [reply Content.Server/Body/Systems/InternalsSystem.cs:197]: okay, got confused, what i actually meant is that Add returns a bool that you should handle

Comment (Plykiya) [reply Content.Server/Body/Systems/InternalsSystem.cs:197]: I guess I can do that

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**